driver-core/src/main/com/mongodb/internal/connection/SocketStream.java

```diff
@@ -79,7 +82,9 @@ public void open() {
             inputStream = socket.getInputStream();
         } catch (IOException e) {
             close();
-            throw new MongoSocketOpenException("Exception opening socket", getAddress(), e);
+            throw translateInterruptedException(e, "Interrupted while connecting")
+                    .<MongoException>map(Function.identity())
+                    .orElseGet(() -> new MongoSocketOpenException("Exception opening socket", getAddress(), e));
```

Comment:
nit:  
We might as well omit this line `.<MongoException>map(Function.identity())` by using: 
```
  throw translateInterruptedException(e, "Interrupted while connecting")
                    .orElseThrow(() -> new MongoSocketOpenException("Exception opening socket", getAddress(), e));
```

Reply:
Nice! Done.